src/mono/msbuild/AndroidAppBuilder/Templates/MonoRunner.java

```diff
@@ -44,25 +45,22 @@ public void onStart() {
         Context context = getContext();
         AssetManager am = context.getAssets();
         String filesDir = context.getFilesDir().getAbsolutePath();
-        String cacheDir = context.getCacheDir().getAbsolutePath ();
+        String cacheDir = context.getCacheDir().getAbsolutePath();
+        String docsDir  = context.getExternalFilesDir(Environment.DIRECTORY_DOCUMENTS).getAbsolutePath();
 
         copyAssetDir(am, "", filesDir);
 
-        // retcode is what Main() returns in C#
-        int retcode = initRuntime(filesDir, cacheDir);
-        WriteLineToInstrumentation("[Mono] Main() returned " + retcode);
-        runOnMainSync (new Runnable() {
+        int retcode = initRuntime(filesDir, cacheDir, docsDir);
+        runOnMainSync(new Runnable() {
             public void run() {
-                finish (retcode, null);
+                Bundle result = new Bundle();
+                result.putInt("return-code", retcode);
+                // Xharness cli expects "test-results-path" with test results
+                result.putString("test-results-path", docsDir + "/testResults.xml");
```

Comment:
nit: do a File.Exists check to handle cases where the test runner crashes etc. so the xharness client can detect this condition and not try to fetch a non-existing file.

Reply:
xharness uninstall the app automatically after tests (even if the app crashes).
but yeah, I'll fix it anyway